src/predict.rs

```diff
@@ -80,10 +80,9 @@ pub fn pred_h<D: Dim>(output: &mut [u16], stride: usize, left: &[u16]) {
   }
 }
 
-pub fn pred_v(output: &mut [u16], stride: usize, above: &[u16], bh: usize) {
-    let bw = above.len();
-    for line in output.chunks_mut(stride).take(bh) {
-        line[..bw].clone_from_slice(above)
+pub fn pred_v<D: Dim>(output: &mut [u16], stride: usize, above: &[u16]) {
+    for line in output.chunks_mut(stride).take(D::H) {
+        line[..D::W].clone_from_slice(above)
```

Comment:
`above` would enjoy the same change.

Reply:
I'm not sure I understand this comment.

Reply:
I was thinking about passing `&above[..D::W]`

Reply:
Oh I get it, or else there's a copy operation. Bench seems to indicate a slight speed up from this.